velox/type/ShortDecimal.h

```diff
@@ -57,9 +60,21 @@ struct ShortDecimal {
     return unscaledValue_ >= other.unscaledValue_;
   }
 
+  ShortDecimal& operator+=(const ShortDecimal& other) {
+    unscaledValue_ += other.unscaledValue_;
```

Comment:
Do we need to check for overflow?

Reply:
Probably not. I think the operator type inference will ensure values fit. But doesn't hurt to add. I will add.

Reply:
> I think the operator type inference will ensure values fit. 

Would you explain how this works?

Reply:
We do not check for overflow for integer values right? So ShortDecimal is similar. We just treat them as any regular integers.
By operator type inference, I mean at the high level, for example, sum aggregate ensures the intermediate type is LongDecimal and the values will fit based on the precision, and scale.

Reply:
We didn't check for overflow in SUM, but turns out we should. Presto implementation checks, hence, we need to match. #2010 adds these checks.